ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -0,0 +1,208 @@
+        testsuite.visit(finder)
+        for test in finder.tests:
+            # print (f'"{test.tags}"') # for future reference in order to fetch test tags
+            tests.append(test.name)
+        print(f"\nDone. Found {len(tests)} test cases")
+    except Exception as err:
+        print(err)
+        os.chdir(curr_dir)
+        raise
+    os.chdir(curr_dir)
+    return tests, ref
+
+
+def generate_rf_argument_file(tests, output_filepath):
+    """
+    Writes the RobotFramework argument file containing the test selection args
+    to include the extracted new tests in previous stage of this script.
+    """
+    content = ""
+    for testname in tests:
+        content += f"--test {testname.strip()}\n"
+    try:
+        with open(output_filepath, "w") as argfile:
+            argfile.write(content)
+    except Exception as err:
+        print("Failed to generate argument file")
+        print(err)
+
+
+def extract_new_test_cases(test_repo, ref_1, ref_2, ref_2_auto, selector_attribute, output_argument_file):
+    """
+    Wrapping function for all the new tests extraction stages.
+    """
+    repo_local_path, cloned = get_repository(test_repo)
+    tests_1, _ = extract_test_cases_from_ref(repo_local_path, ref_1)
+    tests_2, ref_2 = extract_test_cases_from_ref(repo_local_path, ref_2, ref_2_auto, selector_attribute, ref_1)
+    print("\n---| Computing differences |----")
+    new_tests = list(set(tests_1) - set(tests_2))
+    if len(new_tests) == 0:
+        print(f"[WARN] Done. No new tests found in {ref_1} with respect to {ref_2}!")
+        print("Skip argument file creation")
+    else:
+        print(f"Done. Found {len(new_tests)} new tests in {ref_1} which were not present in {ref_2}")
+        if output_argument_file is not None:
+            print("\n---| Generating RobotFramework arguments file |----")
+            generate_rf_argument_file(new_tests, output_argument_file)
+            print("Done.")
+    if cloned:
+        print(f"\n---| Deleting cloned repo in {repo_local_path} |----")
+        shutil.rmtree(repo_local_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        usage=argparse.SUPPRESS,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Script to fetch newly added test cases",
+    )
+
+    parser.add_argument(
+        "-A",
+        "--output-argument-file",
+        help="generate argument file for RobotFramework to include test cases. It expects to receive a file path",
+        action="store",
+        dest="output_argument_file",
+        default=None,
+    )
+    parser.add_argument(
+        "--test-repo",
+        help="ODS-CI repository. It accepts either local path or URL",
+        action="store",
+        dest="test_repo",
+        default="https://github.com/red-hat-data-services/ods-ci",
+    )
+    parser.add_argument(
+        "--ref1",
+        help="first branch or commit to use for comparison (e.g., older one)",
```

Comment:
Hi @bdattoma from the example at the top of this script 
```
--ref1 releases/2.8.0 --ref2-auto true
Done. Found 30 new tests in releases/2.8.0 which were not present in origin/releases/2.7.0
```
it seems that `ref1` is the new branch and `ref2` is the older one (which is opposite to this help)

Reply:
yes ref1 is the newer one.

> which is opposite to this help


why?